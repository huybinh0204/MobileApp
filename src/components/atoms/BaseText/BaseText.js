import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import Animated from 'react-native-reanimated';

const BaseText = ({ children, variant, ...props }) => {
  const TextComponent = variant === 'animated' ? Animated.Text : Text;
  return <TextComponent {...props}>{children}</TextComponent>;
};

BaseText.propTypes = {
  allowFontScaling: PropTypes.bool,
  variant: PropTypes.any,
};

BaseText.defaultProps = {
  allowFontScaling: false,
  variant: 'normal',
};

export default BaseText;
