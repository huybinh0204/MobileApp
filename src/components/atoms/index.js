export { default as AccountNumber } from './AccountNumber/AccountNumber';
export { default as Badge } from './Badge/Badge';
export { default as Balance } from './Balance/Balance';
export { default as BaseFormInput } from './BaseFormInput/BaseFormInput';
export { default as BaseText } from './BaseText/BaseText';
export { default as BottomSheet } from './BottomSheet/BottomSheet';
export { default as Card } from './Card/Card';
export { default as CurrencyInput } from './CurrencyInput/CurrencyInput';
export { default as Dash } from './Dash/Dash';
export { default as Form } from './Form/Form';
export { default as FormInput } from './FormInput/FormInput';
export { default as FormSubmitButton } from './FormSubmitButton/FormSubmitButton';
export { default as GeneralContainer } from './GeneralContainer/GeneralContainer';
export { default as IconSvg } from './IconSvg/IconSvg';
export { default as Link } from './Link/Link';
export { default as MainButton } from './MainButton/MainButton';
export { default as Money } from './Money/Money';
export { default as PdfView } from './PdfView/PdfView';
export { default as Plaid } from './Plaid/Plaid';
export { default as PressableListItem } from './PressableListItem/PressableListItem';
export { default as ProgressBar } from './ProgressBar/ProgressBar';
export { default as SegmentedTextInput } from './SegmentedTextInput/SegmentedTextInput';
export { default as SelectButton } from './SelectButton/SelectButton';
export { default as Switch } from './Switch/Switch';
export { default as TabItem } from './TabItem/TabItem';
export { default as Toast } from './Toast/Toast';
export { default as TransactionItem } from './TransactionItem/TransactionItem';
