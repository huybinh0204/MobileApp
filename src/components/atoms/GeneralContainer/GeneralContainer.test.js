import { render } from '@testing-library/react-native';
import React from 'react';
import { GeneralContainer } from '_components/atoms';
import BaseText from '_components/atoms/BaseText/BaseText';
import Theme from '_utilities/Theme';

describe('GeneralContainer', () => {
  describe('structure', () => {
    it('should match the snapshot for GeneralContainer', () => {
      const { toJSON } = render(
        <Theme>
          <GeneralContainer>
            <BaseText>content</BaseText>,
          </GeneralContainer>
        </Theme>
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
