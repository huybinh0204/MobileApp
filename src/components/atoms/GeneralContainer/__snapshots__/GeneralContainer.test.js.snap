// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralContainer structure should match the snapshot for GeneralContainer 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FAFAFA",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
  >
    content
  </Text>
  ,
</RCTSafeAreaView>
`;
